pkg/api/store/workload/workload_store.go

```diff
@@ -143,9 +148,6 @@ func setSelector(schemaID string, data map[string]interface{}) {
 }
 
 func setSecrets(apiContext *types.APIContext, data map[string]interface{}) {
-	if val, _ := values.GetValue(data, "imagePullSecrets"); val != nil {
```

Comment:
I think removing this line would result in inability to reset image pull secrets to empty. Could you validate this use case.

Reply:
No, since `data` already has the updated values - I can set `imagePullSecrets` empty in API, and reflects correctly

Reply:
What if user set his own imagePullSecrets using kubectl? Can we test the case when such workload created using kubectl, and then edit->save with no changes in the UI, and see if the secrets are preserved.

Reply:
tested, works as expected